std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
```

Comment:
I wish we would just finally settle the issue of autodecoding once and for all and go one way or another, instead of adding more and more redundancy between autodecoding and non-autodecoding functions in Phobos.